calculator.c

```
@@ -0,0 +1,120 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define max 100
+
+// Below is  function for creating a stack and all its operations like initializing a stack, push(), pop()
+typedef struct{
+    int top;
+    int elements[max];
+} stack;
+
+void init_stack(stack *s)
+{
+    s->top = -1;
+}
+
+void push(stack *s, int value)
+{
+    if(s->top == max-1)
+    {
+        printf("Overflow");
+        exit(1);
+    }
+
+    else
+    s->elements[++(s->top)] = value;
+}
+
+int pop(stack *s)
+{
+    if(s->top == -1)
+    {
+        printf("Underflow Condition");
+        exit(1);

```

exit should not be used, you can return an error code and then terminate the main function